service/resource_router_test.go

```diff
@@ -0,0 +1,195 @@
+			AccessKeyID:     "accessKeyID",
+			AccessKeySecret: "accessKeySecret",
+			SessionToken:    "sessionToken",
+		}
+	}
+
+	var certWatcher *certificatetpr.Service
+	{
+		certConfig := certificatetpr.DefaultServiceConfig()
+		certConfig.K8sClient = fake.NewSimpleClientset()
+		certConfig.Logger = microloggertest.New()
+		certWatcher, err = certificatetpr.NewService(certConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	var keyWatcher *randomkeytpr.Service
+	{
+		keyConfig := randomkeytpr.DefaultServiceConfig()
+		keyConfig.K8sClient = fake.NewSimpleClientset()
+		keyConfig.Logger = microloggertest.New()
+		keyWatcher, err = randomkeytpr.NewService(keyConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	var ccService *cloudconfig.CloudConfig
+	{
+		ccServiceConfig := cloudconfig.DefaultConfig()
+		ccServiceConfig.Logger = microloggertest.New()
+		ccServiceConfig.Logger = microloggertest.New()
+
+		ccService, err = cloudconfig.New(ccServiceConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	var legacyResource framework.Resource
+	{
+		legacyConfig := legacyresource.DefaultConfig()
+		legacyConfig.AwsConfig = awsConfig
+		legacyConfig.AwsHostConfig = awsConfig
+		legacyConfig.CertWatcher = certWatcher
+		legacyConfig.CloudConfig = ccService
+		legacyConfig.InstallationName = "test"
+		legacyConfig.K8sClient = fake.NewSimpleClientset()
+		legacyConfig.KeyWatcher = keyWatcher
+		legacyConfig.Logger = microloggertest.New()
+		legacyConfig.PubKeyFile = "test"
+
+		legacyResource, err = legacyresource.New(legacyConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	var cloudformationResource framework.Resource
+	{
+		cloudformationConfig := cloudformationresource.DefaultConfig()
+		cloudformationConfig.Logger = microloggertest.New()
+
+		cloudformationResource, err = cloudformationresource.New(cloudformationConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	namespaceConfig := namespaceresource.DefaultConfig()
+	namespaceConfig.K8sClient = fake.NewSimpleClientset()
+	namespaceConfig.Logger = microloggertest.New()
+
+	namespaceResource, err := namespaceresource.New(namespaceConfig)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	allResources := []framework.Resource{
+		legacyResource,
+		cloudformationResource,
+		namespaceResource,
+	}
+	legacyResources := []framework.Resource{
+		legacyResource,
+	}
+
+	versionedResources := make(map[string][]framework.Resource)
+	versionedResources["0.1.0"] = legacyResources
+	versionedResources["0.2.0"] = allResources
+
+	testCases := []struct {
+		customObject      awstpr.CustomObject
+		expectedResources []framework.Resource
+		errorMatcher      func(err error) bool
+		resourceRouter    func(ctx context.Context, obj interface{}) ([]framework.Resource, error)
+	}{
+		// Case 1. No version in version bundle so return legacy resources.
```

Comment:
FYI we started to count tests starting with 0 in some places.

Reply:
Nice that simplifies the tests. I'll fix in my next PR to do the resource versioning.